### ISSUE TITLE
HPOS break transient function

### ISSUE THREAD

**Issue by dans-art** [woothemes/woocommerce#39279]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I wrote a plugin that allows you to add a customer when creating a new order.
-> https://github.com/dans-art/wac-woocommerce-add-customer
I updated the code and Version 1.7 works now, except for the messages after the order update.
For saving the order I use the hook:

```
add_action('woocommerce_process_shop_order_meta', [$this, 'wac_save_order'], 99, 2);
 public function wac_save_order($order_id, $posted)
    {
        //Load the order
        $order = wc_get_order($order_id);
        if (!$order) {
            $this->log_event_message(__('No Order found', 'wac'), $order_id, 'error');
            return;
        }
        if (isset($_REQUEST['wac_add_customer']) and $_REQUEST['wac_add_customer'] == 'true') {
            //Add new customer
            $email = isset($_REQUEST['_billing_email']) ? sanitize_email($_REQUEST['_billing_email']) : false;
            $existing_user = get_user_by('email', $email);
            $user_id = null;
            if ($existing_user === false) {
                $new_customer_id = $this->wac_add_customer($email, $order_id);
                if ($new_customer_id) {
                    //Link the Order to the Customer
                    $_REQUEST['customer_user'] = $new_customer_id;
                    $order->set_customer_id($new_customer_id);
                    $user_id = $new_customer_id;
                }
            } else {
                //Link Order to Customer
                $this->log_event("existing_account", $order_id, $email);
                $_REQUEST['customer_user'] = $existing_user->ID;
                $user_id =  $existing_user->ID;
                $order->set_customer_id($user_id);
            }
            do_action('wac_after_insert_new_customer', $user_id, $order_id);
            $order->save();
            return true;
        }
        if (isset($_REQUEST['wac_update_customer']) and $_REQUEST['wac_update_customer'] == 'true') {
            //Update the customer Data
            $user_id = $order->get_customer_id();
            if ($user_id === false or empty($user_id)) {
                $this->log_event("no_user_id", $order_id);
                return false;
            }
            $update_customer = $this->wac_add_customer_meta($user_id, $order_id);
            if ($update_customer > 0) {
                $this->log_event("customer_updated", $order_id, $update_customer, $user_id);
                $this->increase_wac_counter('edit');
            }
            do_action('wac_after_insert_updated_customer', $user_id, $order_id);
        }
        return;
    }
```

The function log_event sets the transient. And the hook admin_notices displays the same afterwards.

```
add_action('admin_notices', [$this, 'wac_display_notices'], 10000);
public function wac_display_notices()
    {        
        error_log("Transient read");
        $user_id = get_current_user_id();
        if(isset($_GET['id'])){
            $order_id = $_GET['id'];
        }
        else if(isset($_GET['post'])){
            $order_id = $_GET['post'];
        }
        else{
            return;
        }
        $trans_id = "wac_admin_notice_{$user_id}_{$order_id}";
        $notices = get_transient($trans_id);
        if (is_array($notices)) {
            foreach ($notices as $notice) {
                echo $notice;
            }
        }
        delete_transient($trans_id);
    }
```

When HPSO is activated, the admin message is not show, but without, it works...
Any Ideas?

### Expected behavior

` add_action('admin_notices', [$this, 'wac_display_notices'], 10000;`
Should display the messages for the admin

### Actual behavior

No message is displayed.

### Steps to reproduce

1. Install Version 1.7 of the Plugin: https://github.com/dans-art/wac-woocommerce-add-customer
2. Try it with and without HPOS activated

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://localhost/wordpress
Site address (URL): https://localhost/wordpress
WC Version: 7.8.2
REST API Version: ✔ 7.8.2
WC Blocks Version: ✔ 10.2.4
Action Scheduler Version: ✔ 3.5.4
Log Directory Writable: ✔
WP Version: 6.2.2
WP Multisite: –
WP Memory Limit: 4 GB
WP Debug Mode: –
WP Cron: ✔
Language: de_DE
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.54 (Win64) OpenSSL/1.1.1p PHP/8.0.25
PHP Version: 8.0.25
PHP Post Max Size: 390 MB
PHP Time Limit: 240
PHP Max Input Vars: 1000
cURL Version: 7.76.1
OpenSSL/1.1.1p

SUHOSIN Installed: –
MySQL Version: 10.4.27-MariaDB
Max Upload Size: 390 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.8.2
WC Database Prefix: dawp_
Datenbank-Gesamtgröße: 106.12MB
Datenbank-Datengröße: 94.83MB
Datenbank-Indexgröße: 11.29MB
dawp_woocommerce_sessions: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_woocommerce_api_keys: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_woocommerce_attribute_taxonomies: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_woocommerce_downloadable_product_permissions: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
dawp_woocommerce_order_items: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_woocommerce_order_itemmeta: Daten: 0.14MB + Index: 0.16MB + Engine InnoDB
dawp_woocommerce_tax_rates: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
dawp_woocommerce_tax_rate_locations: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_woocommerce_shipping_zones: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_woocommerce_shipping_zone_locations: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_woocommerce_shipping_zone_methods: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_woocommerce_payment_tokens: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_woocommerce_payment_tokenmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_woocommerce_log: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_2_commentmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_2_comments: Daten: 0.02MB + Index: 0.08MB + Engine InnoDB
dawp_2_links: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_2_loginizer_logs: Daten: 0.00MB + Index: 0.00MB + Engine MyISAM
dawp_2_options: Daten: 0.28MB + Index: 0.02MB + Engine InnoDB
dawp_2_postmeta: Daten: 0.23MB + Index: 0.22MB + Engine InnoDB
dawp_2_posts: Daten: 0.48MB + Index: 0.06MB + Engine InnoDB
dawp_2_termmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_2_terms: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_2_term_relationships: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_2_term_taxonomy: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_2_usage_access: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_2_usage_ref: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_2_yoast_seo_links: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_2_yoast_seo_meta: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_3_loginizer_logs: Daten: 0.00MB + Index: 0.00MB + Engine MyISAM
dawp_actionscheduler_actions: Daten: 0.05MB + Index: 0.13MB + Engine InnoDB
dawp_actionscheduler_claims: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_actionscheduler_groups: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_actionscheduler_logs: Daten: 0.05MB + Index: 0.03MB + Engine InnoDB
dawp_commentmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_comments: Daten: 0.14MB + Index: 0.09MB + Engine InnoDB
dawp_ea_error_logs: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_ewwwio_images: Daten: 2.52MB + Index: 1.92MB + Engine InnoDB
dawp_ewwwio_queue: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_finaltiles_gallery: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_finaltiles_gallery_images: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_fusion_forms: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_fusion_form_entries: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_fusion_form_fields: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_fusion_form_submissions: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_kanban_boards: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_kanban_estimates: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_kanban_log_comments: Daten: 0.06MB + Index: 0.00MB + Engine InnoDB
dawp_kanban_log_status_changes: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_kanban_options: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_kanban_projects: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_kanban_statuses: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_kanban_taskmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_kanban_tasks: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_kanban_task_hours: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_layerslider: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_layerslider_revisions: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_links: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_loginizer_logs: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_lrsync: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_lrsync_collections: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_lrsync_meta: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_lrsync_relations: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
dawp_mgmlp_folders: Daten: 0.00MB + Index: 0.00MB + Engine MyISAM
dawp_newsletter: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_newsletter_emails: Daten: 0.11MB + Index: 0.00MB + Engine InnoDB
dawp_newsletter_sent: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_newsletter_stats: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_newsletter_user_logs: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_options: Daten: 4.34MB + Index: 0.14MB + Engine InnoDB
dawp_postmeta: Daten: 7.52MB + Index: 4.03MB + Engine InnoDB
dawp_posts: Daten: 22.52MB + Index: 0.66MB + Engine InnoDB
dawp_revslider_css: Daten: 0.13MB + Index: 0.02MB + Engine InnoDB
dawp_revslider_css_bkp: Daten: 0.14MB + Index: 0.00MB + Engine InnoDB
dawp_revslider_layer_animations: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_revslider_layer_animations_bkp: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_revslider_navigations: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_revslider_navigations_bkp: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_revslider_sliders: Daten: 0.17MB + Index: 0.02MB + Engine InnoDB
dawp_revslider_sliders_bkp: Daten: 0.31MB + Index: 0.00MB + Engine InnoDB
dawp_revslider_slides: Daten: 1.52MB + Index: 0.02MB + Engine InnoDB
dawp_revslider_slides_bkp: Daten: 1.52MB + Index: 0.00MB + Engine InnoDB
dawp_revslider_static_slides: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_revslider_static_slides_bkp: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_simple_history: Daten: 0.09MB + Index: 0.06MB + Engine InnoDB
dawp_simple_history_contexts: Daten: 1.02MB + Index: 0.28MB + Engine InnoDB
dawp_termmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_terms: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_term_relationships: Daten: 0.08MB + Index: 0.05MB + Engine InnoDB
dawp_term_taxonomy: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_tutor_earnings: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_tutor_quiz_attempts: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_tutor_quiz_attempt_answers: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_tutor_quiz_questions: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_tutor_quiz_question_answers: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_tutor_withdraws: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_usage_access: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_usage_ref: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_usermeta: Daten: 0.14MB + Index: 0.14MB + Engine InnoDB
dawp_users: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
dawp_wcpdf_invoice_number: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wcpdf_packing_slip_number: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wc_admin_notes: Daten: 0.09MB + Index: 0.00MB + Engine InnoDB
dawp_wc_admin_note_actions: Daten: 0.06MB + Index: 0.02MB + Engine InnoDB
dawp_wc_category_lookup: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wc_customer_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_wc_download_log: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_wc_orders: Daten: 0.02MB + Index: 0.11MB + Engine InnoDB
dawp_wc_orders_meta: Daten: 0.30MB + Index: 0.25MB + Engine InnoDB
dawp_wc_order_addresses: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
dawp_wc_order_coupon_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_wc_order_operational_data: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_wc_order_product_lookup: Daten: 0.05MB + Index: 0.06MB + Engine InnoDB
dawp_wc_order_stats: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
dawp_wc_order_tax_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_wc_product_attributes_lookup: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_wc_product_download_directories: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_wc_product_meta_lookup: Daten: 0.02MB + Index: 0.09MB + Engine InnoDB
dawp_wc_rate_limits: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_wc_reserved_stock: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wc_tax_rate_classes: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_wc_webhooks: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_wfblockediplog: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wfblocks7: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
dawp_wfconfig: Daten: 3.17MB + Index: 0.00MB + Engine InnoDB
dawp_wfcrawlers: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wffilechanges: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wffilemods: Daten: 24.58MB + Index: 0.00MB + Engine InnoDB
dawp_wfhits: Daten: 0.42MB + Index: 0.25MB + Engine InnoDB
dawp_wfhoover: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_wfissues: Daten: 0.13MB + Index: 0.06MB + Engine InnoDB
dawp_wfknownfilelist: Daten: 11.52MB + Index: 0.00MB + Engine InnoDB
dawp_wflivetraffichuman: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_wflocs: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wflogins: Daten: 0.45MB + Index: 0.14MB + Engine InnoDB
dawp_wfls_2fa_secrets: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_wfls_role_counts: Daten: 0.00MB + Index: 0.00MB + Engine MEMORY
dawp_wfls_settings: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wfnotifications: Daten: 0.11MB + Index: 0.00MB + Engine InnoDB
dawp_wfpendingissues: Daten: 0.14MB + Index: 0.06MB + Engine InnoDB
dawp_wfreversecache: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wfsnipcache: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
dawp_wfstatus: Daten: 1.03MB + Index: 0.09MB + Engine InnoDB
dawp_wftrafficrates: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_wpml_mails: Daten: 6.52MB + Index: 0.00MB + Engine InnoDB
dawp_yoast_indexable: Daten: 0.30MB + Index: 0.09MB + Engine InnoDB
dawp_yoast_indexable_hierarchy: Daten: 0.05MB + Index: 0.05MB + Engine InnoDB
dawp_yoast_migrations: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
dawp_yoast_primary_term: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
dawp_yoast_seo_links: Daten: 0.11MB + Index: 0.06MB + Engine InnoDB
dawp_yoast_seo_meta: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_zpm_activity: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_zpm_categories: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_zpm_messages: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_zpm_projects: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
dawp_zpm_tasks: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

acf-field: 7
acf-field-group: 3
attachment: 558
avada_faq: 2
avada_portfolio: 7
courses: 1
custom_css: 1
flamingo_contact: 507
flamingo_inbound: 703
foogallery: 24
fusion_element: 9
fusion_template: 14
lesson: 1
mgmlp_media_folder: 25
nav_menu_item: 15
oembed_cache: 15
page: 31
post: 15
product: 32
product_variation: 34
revision: 934
sep_ship_prov: 1
shop_order: 62
shop_order_placehold: 1
shop_order_refund: 1
slide: 8
topics: 1
tutor_enrolled: 1
web-story: 1
wp_global_styles: 1
wp_navigation: 5
wpcf7_contact_form: 8

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (29) ###

Query Monitor: von John Blackbourn – 3.13.1
1A Dan's Dev Helpers: von Dan's Art – 1.0
WooCommerce - Kunde hinzufügen: von Dan's Art – 1.7
Advanced Custom Fields: von WP Engine – 6.1.7
AI Engine: ChatGPT Chatbot, Inhalts-Generator, GPT 3 & 4, extrem anpassbar: von Jordy Meow – 1.8.5
Akismet Anti-Spam: Spam Protection: von Automattic - Anti Spam Team – 5.2
Conditional Fields for Contact Form 7: von Jules Colle – 2.3.9
Contact Form 7: von Takayuki Miyoshi – 5.7.7
Custom Permalinks: von Sami Ahmed Siddiqui – 2.5.2
Enable Media Replace: von ShortPixel – 4.1.2
EWWW Image Optimizer: von Exactly WWW – 7.1.0
Flamingo: von Takayuki Miyoshi – 2.4
FooBox Image Lightbox: von FooPlugins – 2.7.27
FooGallery: von FooPlugins – 2.2.44
Media Cleaner: von Jordy Meow – 6.6.4
Media Usage: von Jordy Meow – 0.0.4
Newsletter: von Stefano Lissa &
Das Newsletter Team – 7.8.8
Responsive Lightbox & Gallery: von dFactory – 2.4.5
Simple History: von Pär Thernström – 4.2.1
Super einfache Lagerverwaltung: von Dan's Art – 1.3.6
Advanced Editor Tools: von Automattic – 5.9.0
Tutor LMS: von Themeum – 2.2.2
Twenty20 Image Before-After: von Zayed Baloch – 1.6.1
PDF Invoices & Packing Slips for WooCommerce: von WP Overnight – 3.5.6
WooCommerce: von Automattic – 7.8.2
WP Fastest Cache: von Emre Vona – 1.1.7
WP Mail Logging: von WP Mail Logging Team – 1.12.0
Media Library Categories: von Jeffrey-WP – 2.0.0
YouTube Embed: von David Artiss – 5.3

### Inactive Plugins (39) ###

Add From Server: von Dion Hulse – 3.4.5
Another Show Hooks: von Exlac – 1.0.2
Avada Builder: von ThemeFusion – 3.5
Avada Core: von ThemeFusion – 5.5
B2B Market: von MarketPress – 1.0.11.1
BackWPup: von Inpsyde GmbH – 4.0.0
Barcode Printing for WooCommerce & Contact Form 7 - (business): von UkrSolution – 3.4.3
BasePress: von codeSavory – 2.15.21
BetterDocs: von WPDeveloper – 2.3.7
DocsPress: von nK – 2.3.1
Easy Appointments: von Nikola Loncar – 3.11.10
Envato Market: von Envato – 2.0.7
Hello Dolly: von Matt Mullenweg – 1.7.2
Kanban Boards for WordPress: von Kanban for WordPress – 2.5.21
Loco Translate: von Tim Whitlock – 2.6.4
Min Max Default Quantity for WooCommerce: von WPFactory – 4.5.9
Photo Engine (WP/LR Sync): von Jordy Meow – 6.2.5
Raiffeisen Email to WooCommerce: von Dan's Art – 0.4
Shopify Actions: von Daniel Spycher – 1
Simple Cloudflare Turnstile: von Elliot Sowersby
RelyWP – 1.22.1

Slider Revolution: von ThemePunch – 6.5.7
Super Easy Picklist for WooCommerce: von Dan's Art – 0.1
Time Tracker for WooCommerce: von DansArt – 1.3.2
Web Stories: von Google – 1.33.0
weDocs: von weDevs – 1.7.1
WooCommerce PayPal Payments: von WooCommerce – 2.1.0
WooCommerce POS: von kilbot – 1.2.4
WooCommerce Product Add-ons: von WooCommerce – 3.0.24
Wordfence Security: von Wordfence – 7.10.1
Wordify: von  – 0.1
WPBakery Page Builder: von Michael M - WPBakery.com – 6.9.0
WP Crontrol: von John Blackbourn & crontributors – 1.15.3
WP Hooks Finder: von Muhammad Rehman – 1.2.3
WPMobile.App — Android and iOS Mobile Application: von WPMobile.App – 11.28
WP Rollback: von GiveWP.com – 1.7.3
Yoast SEO: von Team Yoast – 20.11
YotuWP - YouTube Gallery: von  – 1.3.12
Zass Plugin: von theAlThemist – 2.3.9
Zephyr Project Manager: von Dylan James – 3.3.94

### Dropin Plugins (1) ###

db.php: Query Monitor Database Class (Drop-in)

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: CHF (CHF)
Currency Position: left_space
Thousand Separator: '
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### WC Pages ###

Shop-Basis: #34 - /shop/
Warenkorb: #35 - /warenkorb/
Kasse: #36 - /kasse/
Mein Konto: #37 - /mein-konto/
Allgemeine Geschäftsbedingungen: #9 - /ueber-mich/

### Theme ###

Name: Twenty Twenty-Three Child
Version: 1.0
Author URL: https://wordpress.org
Child Theme: ✔
Parent Theme Name: Twenty Twenty-Three
Parent Theme Version: 1.1
Parent Theme Author URL: https://de.wordpress.org
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: core-profiler
minified-js
new-product-management-experience
product-variation-management
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2023-07-18 17:16:07 +00:00
Options: ✔
Notes: 165
Onboarding: completed

### Action Scheduler ###

Abgeschlossen: 52
Oldest: 2023-06-22 06:31:32 +0000
Newest: 2023-07-17 18:48:15 +0000

Fehlgeschlagen: 14
Oldest: 2023-02-14 20:22:35 +0000
Newest: 2023-06-02 21:27:34 +0000

Ausstehend: 1
Oldest: 2023-07-18 11:49:07 +0000
Newest: 2023-07-18 11:49:07 +0000

### Status report information ###

Generated at: 2023-07-17 18:58:52 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by dans-art** [comment]
I did some testing and it looks like, the hook `admin_notices` runs before` woocommerce_process_shop_order_meta` when HPOS is activated. This leads the the issue that I cannot display a message of the action done within the `woocommerce_process_shop_order_meta ` hook.

**Comment by jorgeatorres** [comment]
Hi @dans-art,

Thanks for reaching out. We tried your code snippet and we found that things seem to work fine because, even though `admin_notices` was running before `woocommerce_process_shop_order_meta`, after the update a redirect would happen that would trigger `admin_notices` once more.

In any case, `woocommerce_process_shop_order_meta` should definitely be executed first, and so we've fixed this in #39590. As such, I'm going to close this issue, but feel free to re-open if I've missed something or if you're still having trouble.

Thanks again!
